File: leetcode/0219-contains-duplicate-ii/containsNearbyDuplicate.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
  let result = false;
  const numIndexMap = new Map<number, number>();

  for (const [index, num] of nums.entries()) {
    if (numIndexMap.has(num) && index - numIndexMap.get(num)! <= k) {
      result = true;
      break;
    }

    numIndexMap.set(num, index);
  }

  return result;
}

export default containsNearbyDuplicate;

File: leetcode/0219-contains-duplicate-ii/containsNearbyDuplicate.go
package leetcode

func containsNearbyDuplicate(nums []int, k int) bool {
	var result = false
	var numIndexMap = map[int]int{}

	for index, num := range nums {

		if leftIndex, ok := numIndexMap[num]; ok && index-leftIndex <= k {
			result = true
			break
		}

		numIndexMap[num] = index
	}

	return result
}
